src/code/audio/audio_thread.c

```diff
@@ -0,0 +1,977 @@
+/**
+ * @file audio_thread.c
+ *
+ * Top-level file that coordinates all audio code on the audio thread.
+ */
+#include "global.h"
+
+AudioTask* AudioThread_UpdateImpl(void);
+void AudioThread_SetFadeOutTimer(s32 seqPlayerIndex, s32 fadeTimer);
+void AudioThread_SetFadeInTimer(s32 seqPlayerIndex, s32 fadeTimer);
+void AudioThread_ProcessCmds(u32 msg);
+void AudioThread_ProcessSeqPlayerCmd(SequencePlayer* seqPlayer, AudioCmd* cmd);
+void AudioThread_ProcessChannelCmd(SequenceChannel* channel, AudioCmd* cmd);
+s32 AudioThread_GetSamplePos(s32 seqPlayerIndex, s32 channelIndex, s32 layerIndex, s32* loopEnd, s32* samplePosInt);
+s32 AudioThread_CountAndReleaseNotes(s32 flags);
+
+AudioTask* AudioThread_Update(void) {
+    return AudioThread_UpdateImpl();
+}
+
+AudioTask* AudioThread_UpdateImpl(void) {
+    static AudioTask* sWaitingAudioTask = NULL;
+    u32 numSamplesRemainingInAi;
+    s32 numAbiCmds;
+    s32 pad;
+    s32 j;
+    s32 sp5C;
+    s16* curAiBuffer;
+    OSTask_t* task;
+    s32 index;
+    u32 sp4C;
+    s32 sp48;
+    s32 i;
+
+    gAudioCtx.totalTaskCount++;
+    if ((gAudioCtx.totalTaskCount % gAudioCtx.audioBufferParameters.specUnk4) != 0) {
+        if (gAudioCustomUpdateFunction != NULL) {
+            gAudioCustomUpdateFunction();
+        }
+
+        if (((gAudioCtx.totalTaskCount % gAudioCtx.audioBufferParameters.specUnk4) + 1) ==
+            gAudioCtx.audioBufferParameters.specUnk4) {
+            return sWaitingAudioTask;
+        }
+
+        return NULL;
+    }
+
+    osSendMesg(gAudioCtx.taskStartQueueP, gAudioCtx.totalTaskCount, OS_MESG_NOBLOCK);
+    gAudioCtx.rspTaskIndex ^= 1;
+    gAudioCtx.curAiBufferIndex++;
+    gAudioCtx.curAiBufferIndex %= 3;
+    index = (gAudioCtx.curAiBufferIndex + 1) % 3;
+
+    // Division converts size to numSamples: 2 channels (left/right) * 2 bytes per sample
+    numSamplesRemainingInAi = osAiGetLength() / (2 * SAMPLE_SIZE);
+
+    if (gAudioCtx.resetTimer < 16) {
+        if (gAudioCtx.numSamplesPerFrame[index] != 0) {
+            osAiSetNextBuffer(gAudioCtx.aiBuffers[index], 2 * gAudioCtx.numSamplesPerFrame[index] * (s32)SAMPLE_SIZE);
+            if (gAudioCtx.aiBuffers[index]) {}
+            if (gAudioCtx.numSamplesPerFrame[index]) {}
+        }
+    }
+
+    if (gAudioCustomUpdateFunction != NULL) {
+        gAudioCustomUpdateFunction();
+    }
+
+    sp5C = gAudioCtx.curAudioFrameDmaCount;
```

Comment:
`dmaCount`? or something similar. It is only used here so should be possible to name I think.